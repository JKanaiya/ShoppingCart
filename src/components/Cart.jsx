const Cart = function ({ cartItems }) {
  return (
    <div>
      {cartItems.length == 0 ? (
        <h2>Your Cart is Empty</h2>
      ) : (
        cartItems.map((item) => (
          <div key={item.id}>
            <h2 role="title">{item.title}</h2>
            <p>{item.description}</p>
            <p>{item.price}</p>
          </div>
        ))
      )}
      {cartItems.length != 0 && (
        <h3>
          {
            cartItems.reduce(
              (previous, current) => ({
                count: 1,
                price:
                  previous.count * previous.price +
                  current.count * current.price,
              }),
              {
                count: 1,
                price: 0,
              },
            ).price
          }
        </h3>
      )}
    </div>
  );
};

export default Cart;
