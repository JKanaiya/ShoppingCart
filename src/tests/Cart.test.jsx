import { describe, it, expect } from "vitest";
import { screen, render } from "@testing-library/react";
import Cart from "../components/Cart";

const cartItems = [
  {
    id: 0,
    title: "TestTitle",
    description: "Jeans",
    price: 10,
    count: 1,
    image: "Testing image",
  },
  {
    id: 1,
    title: "TestTitle2",
    description: "Coat",
    price: 12,
    count: 2,
    image: "Testing image",
  },
];
describe("Cart", () => {
  it("Renders Cart is empty if it is empty", () => {
    render(<Cart cartItems={[]} />);
    expect(
      screen.getByRole("heading", { name: "Your Cart is Empty" }),
    ).toBeInTheDocument();
  });
  it("Renders Cart Items when cartItems passed in", () => {
    render(<Cart cartItems={cartItems} />);
    expect(screen.getByText(/testtitle2/i)).toBeInTheDocument();
  });
  it("Doesnt render a price if the cart is empty", () => {
    render(<Cart cartItems={[]} />);
    expect(screen.queryByRole("priceTotal")).not.toBeInTheDocument();
  });
  it("Renders the price based on the items in the cart and their count", () => {
    render(<Cart cartItems={cartItems} />);
    expect(screen.queryByText(34)).toBeInTheDocument();
  });
});
