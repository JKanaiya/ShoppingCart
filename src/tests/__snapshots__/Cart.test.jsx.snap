// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart > Renders Cart is empty if it is empty 1`] = `
<div>
  <div>
    Cart
  </div>
</div>
`;
